test/integration/scheduler/queue_test.go

```diff
@@ -178,89 +179,185 @@ func TestSchedulingGates(t *testing.T) {
 // TestCoreResourceEnqueue verify Pods failed by in-tree default plugins can be
 // moved properly upon their registered events.
 func TestCoreResourceEnqueue(t *testing.T) {
```

Comment:
Made it table-driven. It'll be the basis for https://github.com/kubernetes/kubernetes/issues/122305.